### PR TITLE
LPS-69775 Build soy

### PR DESCRIPTION
**Pull request by diogo-lins**

### CODE REVIEW

**Comment by thiago-rocha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/main/resources/META-INF/resources/grid_field.js [hunk 1]
```diff
@@ -71,53 +55,60 @@ AUI.add(
 					getValue: function() {
 						var instance = this;
 
-						var gridRowsNode = instance.getGridRowsNode();
```
Comment (thiago-rocha) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/main/resources/META-INF/resources/grid_field.js]: Instead of access the DOM to get the selected value you can use the `value` attribute, since we guarantee the `value` is always updated.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/main/resources/META-INF/resources/grid_field.js [hunk 2]
```diff
@@ -149,19 +170,21 @@ AUI.add(
...
-
 						var rowIndex = target.attr('data-row-index');
 
-						var colValue = target.attr('value');
+						instance._setValueInputHidden(rowIndex, target);
+
+						var value = instance.get('value');
 
-						var currentItem = inputNode.item(rowIndex);
+						if (!value) {
```
Comment (thiago-rocha) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/main/resources/META-INF/resources/grid_field.js]: Instead of check if `!value` you could star the `value` attribute with `{}`.

**Review by thiago-rocha — CHANGES_REQUESTED** [review]

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/main/resources/META-INF/resources/grid_field.js [hunk 3]
```diff
@@ -121,24 +100,20 @@ AUI.add(
...
-					_getGridRowsNode: function() {
+					_getFocusTarget: function() {
 						var instance = this;
 
-						var container = instance.get('container');
-
-						var gridRowsNode = container.all('tbody tr');
-
-						return gridRowsNode;
+						return instance.focusTarget;
```
Comment (thiago-rocha) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/main/resources/META-INF/resources/grid_field.js]: Since the `focusTarget` is a private element it should star with a underscore `_ focusTarget`

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+			'.setValue()',
+			function() {
+				it(
+					'should return empty value if set an empty object value',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({});
+
+						assert.equal(gridField.getValue().A, undefined);
```
Comment (thiago-rocha) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js]: Instead of check for specific values (`A`, `B`) you can just check is `getValue` returns `{}`.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+						assert.equal(gridField.getValue().A, '1');
+
+						assert.equal(gridField.getValue().B, '2');
+
+						done();
+					}
+				);
+
+				it(
+					'should return object values if set an only option',
```
Comment (thiago-rocha) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js]: I don't understanding this test description, could you rephrase it?

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+				it(
+					'should return object values only if exists in the UI',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({A: '1', B: '2', C: '3'});
+
+						assert.isObject(gridField.getValue());
+
+						assert.equal(gridField.getValue().A, '1');
```
Comment (thiago-rocha) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js]: You should compare the `getValue` result with the seted value `{A: '1', B: '2'}`

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+		);
+
+		describe(
+			'.getValue()',
+			function() {
+				it(
+					'should return object value if grid has a value in context',
+					function(done) {
+						gridField = createGridField(
+							{
```
Comment (thiago-rocha) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js]: To make sure the value came from the context you should create the Grid using the attribute `context`
`createGridField(
  {
    context: {
      value: {B: '2'}
    }
  }
)`

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+					}
+				);
+			}
+		);
+
+		describe(
+			'.checkItem()',
+			function() {
+				it(
+					'should click in a radio input and get its value',
```
Comment (thiago-rocha) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js:267]: Add a test where a value is already set and you click in a different options to see if the value was changed

**Review by thiago-rocha — CHANGES_REQUESTED** [review]

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js (hunk 4 above)
Comment (diogo-lins) [reply modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js]: I tried to do this way, but the compare between **{}** and **{}** always will be **false**, because two '{}' are different objects.

**Comment by diogo-lins** [comment]
Updated.

**Comment by diogo-lins** [comment]
Updated.

**Comment by thiago-rocha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by diogo-lins** [comment]
Updated.

**Comment by thiago-rocha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by thiago-rocha** [comment]
Pull request submitted to https://github.com/ctampoya/liferay-portal/pull/71. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
